codegen/src/main/kotlin/tools/samt/codegen/KotlinTypesGenerator.kt

```diff
@@ -0,0 +1,117 @@
+package tools.samt.codegen
+
+class KotlinTypesGenerator : Generator {
+    override val identifier: String = "kotlin-types"
+    override fun generate(generatorParams: GeneratorParams): List<CodegenFile> {
+        generatorParams.packages.forEach {
+            generatePackage(it)
+        }
+        return emittedFiles
+    }
+
+    private val emittedFiles = mutableListOf<CodegenFile>()
+
+    private fun generatePackage(pack: SamtPackage) {
+        if (pack.hasModelTypes()) {
+            val packageSource = buildString {
+                appendLine("package ${pack.qualifiedName}")
+                appendLine()
+
+                pack.records.forEach {
+                    appendRecord(it)
+                }
+
+                pack.enums.forEach {
+                    appendEnum(it)
+                }
+
+                pack.aliases.forEach {
+                    appendAlias(it)
+                }
+
+                pack.services.forEach {
+                    appendService(it)
+                }
+            }
+
+            val filePath = pack.qualifiedName.replace('.', '/') + ".kt"
+            val file = CodegenFile(filePath, packageSource)
+            emittedFiles.add(file)
+        }
+    }
+
+    private fun StringBuilder.appendRecord(record: RecordType) {
+        appendLine("class ${record.name}(")
```

Comment:
Maybe data class would be better?

Reply:
That would be problematic once we implement inheritance